src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatWriter.cs

```diff
@@ -210,13 +211,25 @@ internal void ReflectionWriteCollection(XmlWriterDelegator xmlWriter, object obj
                     IEnumerator nonGenericDictEnumerator = (IEnumerator)new CollectionDataContract.DictionaryEnumerator(((IDictionary)obj).GetEnumerator());
                     enumerator = nonGenericDictEnumerator;
                 }
+                else if (collectionDataContract.Kind == CollectionKind.GenericCollection)
+                {
+                    elementType = collectionDataContract.ItemType;
+                }
                 else
                 {
                     enumeratorType = collectionDataContract.GetEnumeratorMethod.ReturnType;
                 }
 
-                MethodInfo getCurrentMethod = enumeratorType.GetMethod(Globals.GetCurrentMethodName, BindingFlags.Instance | BindingFlags.Public, Array.Empty<Type>());
-                Type elementType = getCurrentMethod.ReturnType;
+                if (elementType == null)
+                {
+                    if (enumeratorType == null)
+                    {
+                        throw new InvalidOperationException("enumeratorType is null.");
```

Comment:
2 observations:
1. Should this be localized?
2. Can this be a more informative message?  I don't think anyone would know what to do if they saw an exception with this message.